### PR TITLE
[DDW-82] Implement TLS support

### PR DESCRIPTION
**Pull request by disassembler**
This PR adds TLS support.

## Todos

- [x] Pass on TLS / server path to cardano-wallet via cardano-launcher
- [x] Enable TLS in local development environment 
- [x] Use `https` instead of `http` in `request.js`: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/api/utils/request.js#L56

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] TBD

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: nix/launcher-config.nix [hunk 1]
```diff
@@ -96,25 +96,46 @@ let
     path.macos64 = "${dataDir}/Logs";
   in path.${os};
 
+  tlsConfig = {
+    ca = {
+      organization = "Daedalus";
+      commonName = "Daedalus Self-Signed Root CA";
+      expiryDays = 3650;
```
Comment (nikolaglumac) [inline nix/launcher-config.nix]: @disassembler why is this set to 3650 days and others are 365 days?

Comment (disassembler) [reply nix/launcher-config.nix]: Convention is CA expiry is longer than cert expiry. In this case it really doesn't matter as we don't save the ca key.

Comment (nikolaglumac) [reply nix/launcher-config.nix]: Got it!

**Comment by alan-mcnicholas** [comment]
The latest build from this PR (daedalus-1.0.0-FC1-cardano-wallet-2020.3.16-mainnet_flight-windows-11787.exe) is unable to launch on Windows; installation succedds but when trying to launch I get the following error message displayed.
```
---------------------------
cardano-launcher.exe
---------------------------
Logs: getDirectoryContents:findFirstFile: does not exist (The system cannot find the path specified.)
---------------------------
OK   
---------------------------
```

The state directory only contains the **Logs** and **Secrets-1.0** folders.
Note - the same error message was visible on build 11783.

**Pull request merged**